server/src-lib/Hasura/GraphQL/Schema/Action.hs

```diff
@@ -1,380 +1,275 @@
-
-
-mkDefinitionList :: AnnotatedObjectType -> HashMap ObjectFieldName PGColumnInfo -> [(PGCol, PGScalarType)]
-mkDefinitionList annotatedOutputType fieldReferences =
-  [ (unsafePGCol $ coerce k, mkPGFieldType k v fieldReferences)
-  | (k, v) <- Map.toList $ _aotAnnotatedFields annotatedOutputType
-  ]
-
-mkFieldMap
-  :: AnnotatedObjectType
-  -> ActionInfo
-  -> HashMap ObjectFieldName PGColumnInfo
-  -> RoleName
-  -> HashMap (G.NamedType,G.Name) ResolveField
-mkFieldMap annotatedOutputType actionInfo fieldReferences roleName =
-  Map.fromList $ fields <> catMaybes relationships
+    ActionInfo actionName outputObject definition permissions comment = actionInfo
+    idFieldName = $$(G.litName "id")
+    idFieldDescription = "the unique id of an action"
+
+-- | Async action's unique id
+actionIdParser
+  :: (MonadSchema n m, MonadError QErr m)
+  => m (Parser 'Both n UnpreparedValue)
+actionIdParser =
+  fmap P.mkParameter <$> P.column (PGColumnScalar PGUUID) (G.Nullability False)
+
+actionOutputFields
+  :: forall m n r. (MonadSchema n m, MonadTableInfo r m, MonadRole r m, Has QueryContext r)
+  => AnnotatedObjectType
+  -> m (Parser 'Output n (RQL.AnnFieldsG UnpreparedValue))
+actionOutputFields outputObject = do
+  let scalarOrEnumFields = map scalarOrEnumFieldParser $ toList $ _otdFields outputObject
+  relationshipFields <- forM (_otdRelationships outputObject) $ traverse relationshipFieldParser
+  let allFieldParsers = scalarOrEnumFields <>
+                        maybe [] (catMaybes . toList) relationshipFields
+      outputTypeName = unObjectTypeName $ _otdName outputObject
+      outputTypeDescription = _otdDescription outputObject
+  pure $ P.selectionSet outputTypeName outputTypeDescription allFieldParsers
+         <&> parsedSelectionsToFields RQL.AFExpression
   where
-    fields =
-      flip map (Map.toList $ _aotAnnotatedFields annotatedOutputType) $
-      \(fieldName, (fieldType, fieldTypeInfo)) ->
-        ( (actionOutputBaseType, unObjectFieldName fieldName)
-        , RFPGColumn $ PGColumnInfo
-          (unsafePGCol $ coerce fieldName)
-          (coerce fieldName)
-          0
-          (PGColumnScalar $ mkPGFieldType fieldName (fieldType, fieldTypeInfo) fieldReferences)
-          (G.isNullable fieldType)
-          Nothing
-        )
-    relationships =
-      flip map (Map.toList $ _aotRelationships annotatedOutputType) $
-      \(relationshipName, relationship) ->
-        let remoteTableInfo = _trRemoteTable relationship
-            remoteTable = _tciName $ _tiCoreInfo remoteTableInfo
-            filterAndLimitM = getFilterAndLimit remoteTableInfo
+    scalarOrEnumFieldParser
+      :: ObjectFieldDefinition (G.GType, AnnotatedObjectFieldType)
+      -> FieldParser n (RQL.AnnFieldG UnpreparedValue)
+    scalarOrEnumFieldParser (ObjectFieldDefinition name _ description ty) =
+      let (gType, objectFieldType) = ty
+          fieldName = unObjectFieldName name
+          -- FIXME? (from master)
```

Comment:
Does anything need to be done here?

Reply:
@rakeshkky What would need to be fixed here? Is it the usage of PGColumnInfo?